google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -181,8 +178,10 @@ JTextPane getWarningMessage() {
   public boolean isModified() {
     CloudSdkService sdkService = CloudSdkService.getInstance();
 
-    return !Paths.get(getCloudSdkDirectory() != null ? getCloudSdkDirectory() : "")
-        .equals(sdkService.getSdkHomePath() != null ? sdkService.getSdkHomePath() : Paths.get(""));
+    String typedDirectory = getCloudSdkDirectory() != null ? getCloudSdkDirectory() : "";
```

Comment:
the getCloudSdkDirectory() method seems to primarily obfuscate what it represents. It should at least mirror its setter and be called getCloudSdkDirectoryText(). Or better yet remove the getter and setter entirely or rename them to getCloudSdkDirectoryFieldText() and setCloudSdkDirectoryFieltText().

Also is it possible for getCloudSdkDirectory to be null?

if not this code could be simplified to just cloudSdkDirectoryField.getText().equals(sdkService.getSdkHomePath()).